packages/evm/contracts/BoostCore.sol

```diff
@@ -178,16 +182,35 @@ contract BoostCore is Ownable, ReentrancyGuard {
         address claimant
     ) public payable nonReentrant {
         BoostLib.Boost storage boost = _boosts[boostId_];
-        if (msg.value < claimFee) revert BoostError.InsufficientFunds(address(0), msg.value, claimFee);
-        _routeClaimFee(boost, referrer_);
+        bytes32 key = _generateKey(boostId_, incentiveId_);
+        IncentiveDisbursalInfo storage incentive = incentives[key];
 
+        // Validate the claimant against the allow list and the validator
         if (!boost.allowList.isAllowed(claimant, data_)) revert BoostError.Unauthorized();
 
         // wake-disable-next-line reentrancy (false positive, function is nonReentrant)
         if (!boost.validator.validate(boostId_, incentiveId_, claimant, data_)) revert BoostError.Unauthorized();
+
+        // Get the balance of the asset before the claim
+        uint256 initialBalance = _getAssetBalance(incentive);
+
+        // Execute the claim
         if (!boost.incentives[incentiveId_].claim(claimant, data_)) {
             revert BoostError.ClaimFailed(claimant, data_);
         }
+
+        // Get the balance of the asset after the claim
+        uint256 finalBalance = _getAssetBalance(incentive);
+
+        // Calculate the change in balance and the protocol fee amount
+        uint256 balanceChange = initialBalance > finalBalance ? initialBalance - finalBalance : 0;
+        uint256 protocolFeeAmount = (balanceChange * incentive.protocolFee) / FEE_DENOMINATOR;
+
+        // Transfer the protocol fee to the protocol fee receiver if applicable
+        if (protocolFeeAmount > 0) {
+            _transferProtocolFee(incentive, protocolFeeAmount);
+        }
```

Comment:
I understand that we are not assessing a distinct claim fee, but since the protocol fee is sent to `protocolFeeReceiver` even on clawback, but what is the point of doing this? Why bother storing the protocol fee in this contract at all and doing all this math if it goes to the `protocolFeeReceiver` anyway?

Reply:
Should be addressed by other comment